swap/swap.go

```diff
@@ -63,6 +62,7 @@ type Swap struct {
 	params           *Params            // economic and operational parameters
 	contract         swap.Contract      // reference to the smart contract
 	honeyPriceOracle HoneyOracle        // oracle which resolves the price of honey (in Wei)
+	logger           log.Logger         // logger for swap related messages and audit trail
```

Comment:
Can we not keep the `swapLogger` package wide and only do the `peerLogger `per instance?

Reply:
i think we can, but i'm not sure i see reason to.

could you please explain why this would be preferable? in general i try to avoid global variables and the like.

Reply:
Because the logger is shared, and is not in any way a property of `Swap`. Then we also don't need to pass it around methods.

Reply:
it's a logger specifically for swap, but yes, maybe it doesn't belong in the struct itself (although if the logpath does, one could argue for this).

it certainly doesn't _need_ to be there.

i've changed it back to a global variable.